docs/java/features/odata/generate-client.md

```diff
@@ -0,0 +1,270 @@
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+</TabItem>
+<TabItem value="v2">
+
+1. Update your `application/pom.xml` file by adding the generator plugin under the `<plugin>` section.
+
+    ```XML
+    <plugin>
+        <groupId>com.sap.cloud.sdk.datamodel</groupId>
+        <artifactId>odata-generator-maven-plugin</artifactId>
+        <!-- Please use the latest version here-->
+        <version>3.18.0</version>
+        <executions>
+            <execution>
+                <id>generate-consumption</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>generate</goal>
+                </goals>
+                <configuration>
+                    <inputDirectory>${project.basedir}/edmx</inputDirectory>
+                    <outputDirectory>${project.build.directory}/vdm</outputDirectory>
+                    <deleteOutputDirectory>true</deleteOutputDirectory>
+                    <packageName>com.mycompany.vdm</packageName>
+                    <defaultBasePath>odata/v4/</defaultBasePath>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+</TabItem>
+</Tabs>
+
+2. Adapt the `<inputDirectory>` to point to the place of your service definition.
+
+1. In case the target folder should be automatically added as a source folder by maven you can leverage the build helper plugin:
+    ```XML
+    <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <executions>
+            <execution>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>add-source</goal>
+                </goals>
+                <configuration>
+                    <sources>
+                        <source>${project.build.directory}/vdm</source>
+                    </sources>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+Now maven will run the generator within the `process-sources` phase which is executed before compile.
+
+The following parameters allow to configure the generator further:
+
+|   Parameter Value |   Explanation |
+|-------------------|---------------|
+|`<inputDirectory>`|Location of the metadata file.
+|`<outputDirectory>`|Location of the output directory for generated sources.
+|`<deleteOutputDirectory>`|Target directory is deleted before every execution of the generator
+|`<packageName>`|Package name for the generated sources
+|`<defaultBasePath>`|Base path of the exposed API
+
+:::note
+Please note that if you use the generator for services other than SAP S/4HANA services, you need to add the parameter `defaultBasePath` to the configuration section, which should provide the base path to the exposed API (e.g odata/v4/).
```

Comment:
Not sure, but what do you think about that: (e.g odata/v4/) => (e.g `odata/v4/`) or even (e.g *odata/v4/*) because it is a value...?